src/hdmf/backends/io.py

```diff
@@ -75,6 +75,10 @@ def read(self, **kwargs):
         return container
 
     @docval({'name': 'container', 'type': Container, 'doc': 'the Container object to write'},
+            {'name': 'number_of_jobs', 'type': int, 'doc': 'Number of jobs to use in parallel during write (only operates on GenericDataChunkIterator-wrapped datasets).', 'default': 1},
+            {'name': 'number_of_jobs', 'type': int,
+         'doc': "Number of jobs to use in parallel during write (only operates on GenericDataChunkIterator-wrapped datasets).",
```

Comment:
It looks like the `number_of_jobs` argument is duplicated here. I assume one of these should be removed.

Reply:
Why is this argument needed on HDMFIO? HDMFIO should only expose arguments that we expect all backends to support. If this argument is specific for ZarrIO then we should probably only add it in hdmf-zarr

Reply:
From the lines

```python
with NWBZarrIO(path=..., mode="w") as io:
    io.write(nwbfile, number_of_jobs=number_of_jobs)
```

leads to

https://github.com/hdmf-dev/hdmf-zarr/blob/d47fc8249e76a371fff990b07218a088e59ea590/src/hdmf_zarr/nwb.py#L14

which since it does not redefine `write` leads to

https://github.com/hdmf-dev/hdmf-zarr/blob/d47fc8249e76a371fff990b07218a088e59ea590/src/hdmf_zarr/backend.py#L200

which then quickly directs to the `super` (which is `HDMFIO`) at

https://github.com/hdmf-dev/hdmf-zarr/blob/d47fc8249e76a371fff990b07218a088e59ea590/src/hdmf_zarr/backend.py#L203

I suppose one way around this could be to set it as an instance attribute on the HDMF-Zarr side; but that feels weird since this is not a property of the `NWBZarrIO` class or object, but is a control parameter over how we want the operation `write` to be performed

Reply:
Or I suppose `allow_extra=True` is what allows it to not be specified at the HDMF level, but nonetheless passed implicitly throughout the nested calls